curvallis/curve_editing/smoothers.py

```diff
@@ -329,3 +329,72 @@ def applySmooth(self, xdata, ydata, xmin, xmax):
 
 
 
+class BSplineSmoother(object):
+    """BSplineSmoother creates an object for smoothing 1D data via B-Spline smoothing"""
+    def __init__(self):
+        pass
+    def applySmooth(self, datax, datay, unknown1, unknown2, detail=0):
+        l = len(datax)  #Sets l to the number of data points
+        t = numpy.linspace(0,1,l-2,endpoint=True)
+        t = numpy.append([0,0,0],t)
+        t = numpy.append(t, [1,1,1])
+        #
+        tck = [t,[datax,datay],3]
+        #
+        #if(detail < (l*2)):
+            #detail = l*2
+        detail = l*5
+        u3 = numpy.linspace(0,1,detail,endpoint=True)
+        #
+        out = scipy.interpolate.splev(u3,tck)
+        newy = []
+        #newy.append(out[1][0])
+        for i in range(l):
+            x = datax[i]
+            for i in range(len(out[0])):
+                if(out[0][i] == x):
+                    newy.append(out[1][i])
+                    break
+                elif(i < len(out[0])-1 and out[0][i] > x):
+                    x1, y1 = out[0][i], out[1][i]
+                    x2, y2 = out[0][i+1], out[1][i+1]
+                    m = (y2-y1)/(x2-x1)
+                    b = -1*((m*x1)-y1)
+                    #print("\n(" + str(x1) + "," + str(y1) + ")")
+                    #print("(" + str(x2) + "," + str(y2) + ")")
+                    #print("Slope: " + str(m))
+                    #print("b = " + str(b))
```

Comment:
Hey @sudo-Eric, this may just be an OCD suggestion from me, but if the code works you could probably remove all of the testing code that is commented out. ;)